scripts/docker-integration-tests/prometheus_maptags/test.sh

```diff
@@ -0,0 +1,115 @@
+# quay.io/m3db/prometheus_remote_client_golang @ v0.4.3
+PROMREMOTECLI_IMAGE=quay.io/m3db/prometheus_remote_client_golang@sha256:fc56df819bff9a5a087484804acf3a584dd4a78c68900c31a28896ed66ca7e7b
+JQ_IMAGE=realguess/jq:1.4@sha256:300c5d9fb1d74154248d155ce182e207cf6630acccbaadd0168e18b15bfaa786
+METRIC_NAME_TEST_TOO_OLD=foo
+METRIC_NAME_TEST_RESTRICT_WRITE=bar
+export REVISION
+
+echo "Pull containers required for test"
+docker pull $PROMREMOTECLI_IMAGE
+docker pull $JQ_IMAGE
+
+echo "Run m3dbnode and m3coordinator containers"
+docker-compose -f ${COMPOSE_FILE} up -d dbnode01
+docker-compose -f ${COMPOSE_FILE} up -d coordinator01
+
+TEST_SUCCESS=false
+
+function defer {
+  if [[ "$TEST_SUCCESS" != "true" ]]; then
+    echo "Test failure, printing docker-compose logs"
+    docker-compose -f ${COMPOSE_FILE} logs
+  fi
+
+  docker-compose -f ${COMPOSE_FILE} down || echo "unable to shutdown containers" # CI fails to stop all containers sometimes
+}
+trap defer EXIT
+
+setup_single_m3db_node
+
+echo "Start Prometheus containers"
+docker-compose -f ${COMPOSE_FILE} up -d prometheus01
+
+function test_prometheus_remote_read {
+  # Ensure Prometheus can proxy a Prometheus query
+  echo "Wait until the remote write endpoint generates and allows for data to be queried"
+  ATTEMPTS=50 TIMEOUT=2 MAX_TIMEOUT=4 retry_with_backoff  \
+    '[[ $(curl -sSf 0.0.0.0:9090/api/v1/query?query=prometheus_remote_storage_succeeded_samples_total | jq -r .data.result[0].value[1]) -gt 100 ]]'
+}
+
+function prometheus_remote_write {
+  local metric_name=$1
+  local datapoint_timestamp=$2
+  local datapoint_value=$3
+  local expect_success=$4
+  local expect_success_err=$5
+  local expect_status=$6
+  local expect_status_err=$7
+  local metrics_type=$8
+  local metrics_storage_policy=$9
+
+  network=$(docker network ls --format '{{.ID}}' | tail -n 1)
+  out=$((docker run -it --rm --network $network           \
+    $PROMREMOTECLI_IMAGE                                  \
+    -u http://coordinator01:7201/api/v1/prom/remote/write \
+    -t __name__:${metric_name}                            \
+    -h "M3-Metrics-Type: ${metrics_type}"                 \
+    -h "M3-Storage-Policy: ${metrics_storage_policy}"     \
+    -h 'M3-Map-Tags-By-JSON: {"tagMappers":[{"append":{"tag":"globaltag","value":"somevalue"}}]}' \
+    -d ${datapoint_timestamp},${datapoint_value} | grep -v promremotecli_log) || true)
+  success=$(echo $out | grep -v promremotecli_log | docker run --rm -i $JQ_IMAGE jq .success)
+  status=$(echo $out | grep -v promremotecli_log | docker run --rm -i $JQ_IMAGE jq .statusCode)
+  if [[ "$success" != "$expect_success" ]]; then
+    echo $expect_success_err
+    return 1
+  fi
+  if [[ "$status" != "$expect_status" ]]; then
+    echo "${expect_status_err}: actual=${status}"
+    return 1
+  fi
+  echo "Returned success=${success}, status=${status} as expected"
+  return 0
+}
+
+function prometheus_query_native {
```

Comment:
Would it be worth pulling these tests into an existing prom remote write test? Then can avoid overhead in setting up the stack

Reply:
Agreed, it is starting to get copy/pasted into a lot of different places. Unfortunately bash does tend to trend towards this pattern.. at least the way we're writing it/etc since we're not paying a lot of attention/have a lot of experience with DRY bash.

Reply:
My take though too: would be great if we could actually move this into `common.sh`.